src/mx/TextFileSource.java

```diff
@@ -0,0 +1,45 @@
+package mx;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class TextFileSource implements SourceInterface {
+
+    protected ArrayList<Domain> domains;
+
+    protected String filename;
+
+    public TextFileSource(String inputFilename) {
+        domains = new ArrayList<>();
+        filename = inputFilename;
```

Comment:
Some kind of processing or validation should be made here, to ensure object of this class cannot be created in invalid state - try to open a stream or make sure the file exists. This should fail fast